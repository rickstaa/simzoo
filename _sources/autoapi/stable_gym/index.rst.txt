stable_gym
==========

.. py:module:: stable_gym

.. autoapi-nested-parse::

   Entry_point that register the Stable Gym gymnasium environments.



Subpackages
-----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/common/index
   /autoapi/stable_gym/envs/index


Submodules
----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/version/index


Attributes
----------

.. autoapisummary::

   stable_gym.__version__
   stable_gym.__version_tuple__
   stable_gym.ENVS


Classes
-------

.. autoapisummary::

   stable_gym.MaxEpisodeStepsInjectionWrapper


Package Contents
----------------

.. py:class:: MaxEpisodeStepsInjectionWrapper(env)

   Bases: :py:obj:`gymnasium.Wrapper`


   A gymnasium wrapper that injects the ``max_episode_steps`` attribute of the
   ``TimeLimit`` wrapper into the base environment as the
   ``_time_limit_max_episode_steps`` attribute. If the environment is not wrapped in
   a ``TimeLimit`` wrapper, then the ``_time_limit_max_episode_steps`` attribute is
   set to ``None``.

   Wrap a gymnasium environment.
   :param env: The gymnasium environment.
   :type env: gym.Env


   .. py:attribute:: max_episode_steps


.. py:data:: __version__
   :value: '0.20.7'


.. py:data:: __version_tuple__

.. py:data:: ENVS

