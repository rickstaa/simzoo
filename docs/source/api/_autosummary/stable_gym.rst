stable\_gym
===========

.. automodule:: stable_gym
  
   
   
   

   
   
   

   
   
   

   
   
   



.. rubric:: Modules

.. autosummary::
   :toctree:
   :template: custom-module-template.rst
   :recursive:

   stable_gym.common
   stable_gym.envs

