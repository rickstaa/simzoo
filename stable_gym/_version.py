# Stores the package version number so that it can be accessed from other modules.
version = "0.6.4"
version_tuple = version.split(".")
