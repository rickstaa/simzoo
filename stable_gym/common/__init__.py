"""Contains several functions that can be used across all the Stable Gym gymnasium
environments.
"""
from stable_gym.common.disturber import Disturber  # noqa: F401
