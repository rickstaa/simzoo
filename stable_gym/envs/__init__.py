"""The Stable Gym gymnasium environments."""
