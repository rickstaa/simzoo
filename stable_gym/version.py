# Stores the package version number so that it can be accessed from other modules.
__version__ = "0.6.38"
__version_tuple__ = __version__.split(".")
