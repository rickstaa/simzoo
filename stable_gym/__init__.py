"""Module that register the Stable Gym gymnasium environments."""
import gymnasium as gym
from gymnasium.envs.registration import register

# Make module version available.
from .version import __version__, __version_tuple__

# Available environments.
# TODO: Update reward thresholds.
ENVS = {
    "Oscillator-v1": {
        "module": "stable_gym.envs.biological.oscillator.oscillator:Oscillator",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "OscillatorComplicated-v1": {
        "module": "stable_gym.envs.biological.oscillator_complicated.oscillator_complicated:OscillatorComplicated",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "CartPoleCost-v1": {
        "module": "stable_gym.envs.classic_control.cartpole_cost.cartpole_cost:CartPoleCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Ex3EKF-v1": {
        "module": "stable_gym.envs.classic_control.ex3_ekf.ex3_ekf:Ex3EKF",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "AntCost-v1": {
        "module": "stable_gym.envs.mujoco.ant_cost.ant_cost:AntCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "HalfCheetahCost-v1": {
        "module": "stable_gym.envs.mujoco.half_cheetah_cost.half_cheetah_cost:HalfCheetahCost",
        "max_step": 200,
        "reward_threshold": 300,
    },
    "HopperCost-v1": {
        "module": "stable_gym.envs.mujoco.hopper_cost.hopper_cost:HopperCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "SwimmerCost-v1": {
        "module": "stable_gym.envs.mujoco.swimmer_cost.swimmer_cost:SwimmerCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Walker2dCost-v1": {
        "module": "stable_gym.envs.mujoco.walker2d_cost.walker2d_cost:Walker2dCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
}

for env, val in ENVS.items():
    if env not in gym.envs.registry:  # NOTE: Required because we use namespace packages
        register(
            id=env,
            entry_point=val["module"],
            max_episode_steps=val["max_step"],
            reward_threshold=val["reward_threshold"],
        )
