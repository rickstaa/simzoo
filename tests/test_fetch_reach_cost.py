"""Test if the FetchReachCost environment still behaves like the original FetchReach
environment when the same environment parameters are used.
"""
import gymnasium as gym
import numpy as np
from gymnasium.logger import ERROR

import stable_gym  # noqa: F401

gym.logger.set_level(ERROR)


class TestFetchReachCostEqual:
    # Make original FetchReach environment.
    env = gym.make("FetchReach")
    # Make FetchReachCost environment.
    env_cost = gym.make("FetchReachCost")

    def test_equal_reset(self):
        """Test if reset behaves the same."""
        # Perform reset and check if observations are equal.
        observation, _ = self.env.reset(seed=42)
        observation_cost, _ = self.env_cost.reset(seed=42)
        observation = gym.spaces.flatten(self.env.observation_space, observation)
        observation_cost = gym.spaces.flatten(
            self.env_cost.observation_space, observation_cost
        )

        assert np.allclose(
            observation, observation_cost
        ), f"{observation} != {observation_cost}"

    def test_equal_steps(self):
        """Test if steps behave the same."""
        # Perform several steps and check if observations are equal.
        self.env.reset(seed=42), self.env_cost.reset(seed=42)
        for _ in range(10):
            self.env.action_space.seed(42)
            action = self.env.action_space.sample()
            observation, _, _, _, _ = self.env.step(action)
            observation_cost, _, _, _, _ = self.env_cost.step(action)
            observation = gym.spaces.flatten(self.env.observation_space, observation)
            observation_cost = gym.spaces.flatten(
                self.env_cost.observation_space, observation_cost
            )

            assert np.allclose(
                observation, observation_cost
            ), f"{observation} != {observation_cost}"

    def test_snapshot(self, snapshot):
        """Test if the 'FetchReachCost' environment is still equal snapshot."""
        observation, info = self.env_cost.reset(seed=42)
        observation = gym.spaces.flatten(self.env.observation_space, observation)
        assert (observation == snapshot).all()
        assert info == snapshot
        self.env_cost.action_space.seed(42)
        for _ in range(5):
            action = self.env_cost.action_space.sample()
            observation, reward, terminated, truncated, info = self.env_cost.step(
                action
            )
            observation = gym.spaces.flatten(self.env.observation_space, observation)
            assert (observation == snapshot).all()
            assert reward == snapshot
            assert terminated == snapshot
            assert truncated == snapshot
            assert info == snapshot
