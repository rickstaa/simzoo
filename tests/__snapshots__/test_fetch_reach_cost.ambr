# serializer version: 1
# name: TestFetchReachCostEqual.test_snapshot
  1.3418548590918855
# ---
# name: TestFetchReachCostEqual.test_snapshot.1
  0.7491005080081806
# ---
# name: TestFetchReachCostEqual.test_snapshot.10
  6.923773352298313e-05
# ---
# name: TestFetchReachCostEqual.test_snapshot.100
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.101
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.102
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.103
  1.3202916402758909
# ---
# name: TestFetchReachCostEqual.test_snapshot.104
  0.7677331861111492
# ---
# name: TestFetchReachCostEqual.test_snapshot.105
  0.5829533067486501
# ---
# name: TestFetchReachCostEqual.test_snapshot.106
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.107
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.108
  0.0012439117294339881
# ---
# name: TestFetchReachCostEqual.test_snapshot.109
  -0.024712123324438853
# ---
# name: TestFetchReachCostEqual.test_snapshot.11
  -3.253367286843025e-06
# ---
# name: TestFetchReachCostEqual.test_snapshot.110
  0.017441183843390908
# ---
# name: TestFetchReachCostEqual.test_snapshot.111
  0.0012272377731442558
# ---
# name: TestFetchReachCostEqual.test_snapshot.112
  0.0003076219335113586
# ---
# name: TestFetchReachCostEqual.test_snapshot.113
  0.12510486664436407
# ---
# name: TestFetchReachCostEqual.test_snapshot.114
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.115
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.116
  dict({
    'is_success': 0.0,
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.12
  -2.1965512969677857e-09
# ---
# name: TestFetchReachCostEqual.test_snapshot.13
  5.165812470673677e-06
# ---
# name: TestFetchReachCostEqual.test_snapshot.14
  4.76882451861503e-06
# ---
# name: TestFetchReachCostEqual.test_snapshot.15
  -2.3181035946876523e-06
# ---
# name: TestFetchReachCostEqual.test_snapshot.16
  dict({
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.17
  1.357925981658545
# ---
# name: TestFetchReachCostEqual.test_snapshot.18
  0.7453176396698278
# ---
# name: TestFetchReachCostEqual.test_snapshot.19
  0.5565930680392518
# ---
# name: TestFetchReachCostEqual.test_snapshot.2
  0.5347072045329273
# ---
# name: TestFetchReachCostEqual.test_snapshot.20
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.21
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.22
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.23
  1.357925981658545
# ---
# name: TestFetchReachCostEqual.test_snapshot.24
  0.7453176396698278
# ---
# name: TestFetchReachCostEqual.test_snapshot.25
  0.5565930680392518
# ---
# name: TestFetchReachCostEqual.test_snapshot.26
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.27
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.28
  0.013848829223197369
# ---
# name: TestFetchReachCostEqual.test_snapshot.29
  -0.0032113507225725296
# ---
# name: TestFetchReachCostEqual.test_snapshot.3
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.30
  0.018902367915659708
# ---
# name: TestFetchReachCostEqual.test_snapshot.31
  8.018899994340644e-05
# ---
# name: TestFetchReachCostEqual.test_snapshot.32
  3.0353723624336032e-05
# ---
# name: TestFetchReachCostEqual.test_snapshot.33
  0.10920827489824483
# ---
# name: TestFetchReachCostEqual.test_snapshot.34
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.35
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.36
  dict({
    'is_success': 0.0,
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.37
  1.3347978145778374
# ---
# name: TestFetchReachCostEqual.test_snapshot.38
  0.7738499161633874
# ---
# name: TestFetchReachCostEqual.test_snapshot.39
  0.5744111031080307
# ---
# name: TestFetchReachCostEqual.test_snapshot.4
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.40
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.41
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.42
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.43
  1.3347978145778374
# ---
# name: TestFetchReachCostEqual.test_snapshot.44
  0.7738499161633874
# ---
# name: TestFetchReachCostEqual.test_snapshot.45
  0.5744111031080307
# ---
# name: TestFetchReachCostEqual.test_snapshot.46
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.47
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.48
  -0.023425623484644574
# ---
# name: TestFetchReachCostEqual.test_snapshot.49
  0.02475853769201211
# ---
# name: TestFetchReachCostEqual.test_snapshot.5
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.50
  0.011247720756417957
# ---
# name: TestFetchReachCostEqual.test_snapshot.51
  0.00039687937269950303
# ---
# name: TestFetchReachCostEqual.test_snapshot.52
  0.0007098209151667878
# ---
# name: TestFetchReachCostEqual.test_snapshot.53
  0.12011633319883185
# ---
# name: TestFetchReachCostEqual.test_snapshot.54
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.55
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.56
  dict({
    'is_success': 0.0,
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.57
  1.3096181659553525
# ---
# name: TestFetchReachCostEqual.test_snapshot.58
  0.7733783511588198
# ---
# name: TestFetchReachCostEqual.test_snapshot.59
  0.567784623471936
# ---
# name: TestFetchReachCostEqual.test_snapshot.6
  1.3418548590918855
# ---
# name: TestFetchReachCostEqual.test_snapshot.60
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.61
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.62
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.63
  1.3096181659553525
# ---
# name: TestFetchReachCostEqual.test_snapshot.64
  0.7733783511588198
# ---
# name: TestFetchReachCostEqual.test_snapshot.65
  0.567784623471936
# ---
# name: TestFetchReachCostEqual.test_snapshot.66
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.67
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.68
  -0.016901383108811938
# ---
# name: TestFetchReachCostEqual.test_snapshot.69
  -0.005267642504836206
# ---
# name: TestFetchReachCostEqual.test_snapshot.7
  0.7491005080081806
# ---
# name: TestFetchReachCostEqual.test_snapshot.70
  -0.008059342090323846
# ---
# name: TestFetchReachCostEqual.test_snapshot.71
  0.0011144128405738989
# ---
# name: TestFetchReachCostEqual.test_snapshot.72
  4.43110939060114e-05
# ---
# name: TestFetchReachCostEqual.test_snapshot.73
  0.1430358659919376
# ---
# name: TestFetchReachCostEqual.test_snapshot.74
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.75
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.76
  dict({
    'is_success': 0.0,
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.77
  1.3162255249084425
# ---
# name: TestFetchReachCostEqual.test_snapshot.78
  0.7926694260924992
# ---
# name: TestFetchReachCostEqual.test_snapshot.79
  0.563277726352563
# ---
# name: TestFetchReachCostEqual.test_snapshot.8
  0.5347072045329273
# ---
# name: TestFetchReachCostEqual.test_snapshot.80
  1.4240418362882101
# ---
# name: TestFetchReachCostEqual.test_snapshot.81
  0.7307640400335378
# ---
# name: TestFetchReachCostEqual.test_snapshot.82
  0.6422863222105468
# ---
# name: TestFetchReachCostEqual.test_snapshot.83
  1.3162255249084425
# ---
# name: TestFetchReachCostEqual.test_snapshot.84
  0.7926694260924992
# ---
# name: TestFetchReachCostEqual.test_snapshot.85
  0.563277726352563
# ---
# name: TestFetchReachCostEqual.test_snapshot.86
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.87
  0.0
# ---
# name: TestFetchReachCostEqual.test_snapshot.88
  0.009341557161459504
# ---
# name: TestFetchReachCostEqual.test_snapshot.89
  0.0175230065351739
# ---
# name: TestFetchReachCostEqual.test_snapshot.9
  0.00020023229392314443
# ---
# name: TestFetchReachCostEqual.test_snapshot.90
  -0.002192581047121982
# ---
# name: TestFetchReachCostEqual.test_snapshot.91
  0.00018709159074847438
# ---
# name: TestFetchReachCostEqual.test_snapshot.92
  0.0006045535205191497
# ---
# name: TestFetchReachCostEqual.test_snapshot.93
  0.14730577735478773
# ---
# name: TestFetchReachCostEqual.test_snapshot.94
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.95
  False
# ---
# name: TestFetchReachCostEqual.test_snapshot.96
  dict({
    'is_success': 0.0,
  })
# ---
# name: TestFetchReachCostEqual.test_snapshot.97
  1.3202916402758909
# ---
# name: TestFetchReachCostEqual.test_snapshot.98
  0.7677331861111492
# ---
# name: TestFetchReachCostEqual.test_snapshot.99
  0.5829533067486501
# ---
